### PR TITLE
Update approvals

### LINKED ISSUE #101: Update Approvals
* Make sure default approval types are there
* Use app name instead of ID
* Display warning when updating a version with approvals

### PR DESCRIPTION
**Pull request by prayerslayer**
Fixes #101

### CODE REVIEW

File: client/lib/application/src/approval-form/approval-form.js [hunk 1]
```diff
@@ -39,7 +39,15 @@ class ApprovalForm extends BaseView {
...
         .actions
         .saveApproval(applicationId, versionId, approval)
-        .then(() => this.actions.fetchApprovals(applicationId, versionId));
+        .then(() => this.actions.fetchApprovals(applicationId, versionId))
+        .catch(err => {
+            this
+            .props
+            .notificationActions
+            .addNotification(
+                `Could not approve version ${version_id} of ${this.data.application.name}. ${err.message}`,
```
Comment (elauria) [inline client/lib/application/src/approval-form/approval-form.js]: should be versionId

File: client/lib/application/src/version-form/version-form.js [hunk 2]
```diff
@@ -96,7 +96,7 @@ class VersionForm extends BaseView {
             .props
             .notificationActions
             .addNotification(
-                `Could not ${verb} version ${version_id} for ${this.data.application.name}. ${err.message}`,
+                `Could not ${verb} version ${version_id} of ${this.data.application.name}. ${err.message}`,
```
Comment (elauria) [inline client/lib/application/src/version-form/version-form.js]: should be camelCase everywhere

File: server/mocks/5000-kio.js [hunk 3]
```diff
@@ -160,6 +160,11 @@ server.put('/apps/:id/versions/:ver', function(req, res) {
             versions[id][ver] = req.body;
             versions[id][ver].application_id = id;
             versions[id][ver].id = ver;
+            // delete approvals for this version
+            if (approvals[id] && approvals[id][ver]) {
+                console.log('deleting approvals for', id, ver);
```
Comment (elauria) [inline server/mocks/5000-kio.js]: do we need this console.log?

Comment (prayerslayer) [inline server/mocks/5000-kio.js]: Not really, was for debugging.

----- Ursprüngliche Nachricht -----
Von: "Emanuel Lauria" <notifications@github.com>
Gesendet: ‎19.‎05.‎2015 18:04
An: "zalando-stups/yourturn" <yourturn@noreply.github.com>
Cc: "Nikolaus Piccolotto" <prayerslayer@gmail.com>
Betreff: Re: [yourturn] Update approvals (#103)

In server/mocks/5000-kio.js:
> @@ -160,6 +160,11 @@ server.put('/apps/:id/versions/:ver', function(req, res) {
>              versions[id][ver] = req.body;
>              versions[id][ver].application_id = id;
>              versions[id][ver].id = ver;
> +            // delete approvals for this version
> +            if (approvals[id] && approvals[id][ver]) {
> +                console.log('deleting approvals for', id, ver);
do we need this console.log?

File: client/lib/application/src/version-form/version-form.js (hunk 2 above)
Comment (prayerslayer) [inline client/lib/application/src/version-form/version-form.js]: Yup, true.

----- Ursprüngliche Nachricht -----
Von: "Emanuel Lauria" <notifications@github.com>
Gesendet: ‎19.‎05.‎2015 18:00
An: "zalando-stups/yourturn" <yourturn@noreply.github.com>
Cc: "Nikolaus Piccolotto" <prayerslayer@gmail.com>
Betreff: Re: [yourturn] Update approvals (#103)

In client/lib/application/src/version-form/version-form.js:
> @@ -96,7 +96,7 @@ class VersionForm extends BaseView {
>              .props
>              .notificationActions
>              .addNotification(
> -                `Could not ${verb} version ${version_id} for ${this.data.application.name}. ${err.message}`,
> +                `Could not ${verb} version ${version_id} of ${this.data.application.name}. ${err.message}`,
should be camelCase everywhere

File: client/lib/application/src/approval-form/approval-form.js (hunk 1 above)
Comment (prayerslayer) [inline client/lib/application/src/approval-form/approval-form.js]: Agree.

----- Ursprüngliche Nachricht -----
Von: "Emanuel Lauria" <notifications@github.com>
Gesendet: ‎19.‎05.‎2015 17:58
An: "zalando-stups/yourturn" <yourturn@noreply.github.com>
Cc: "Nikolaus Piccolotto" <prayerslayer@gmail.com>
Betreff: Re: [yourturn] Update approvals (#103)

In client/lib/application/src/approval-form/approval-form.js:
> @@ -39,7 +39,15 @@ class ApprovalForm extends BaseView {
>          this
>          .actions
>          .saveApproval(applicationId, versionId, approval)
> -        .then(() => this.actions.fetchApprovals(applicationId, versionId));
> +        .then(() => this.actions.fetchApprovals(applicationId, versionId))
> +        .catch(err => {
> +            this
> +            .props
> +            .notificationActions
> +            .addNotification(
> +                `Could not approve version ${version_id} of ${this.data.application.name}. ${err.message}`,
should be versionId

**Pull request merged**